### PR TITLE
Added last_altered query to Snowflake catalog macro

### PR DESCRIPTION
**Pull request by Mr-Nobody99**
resolves ##2728

### Description
Added addition column using snowflakes last_altered col of information schema, which should allow dbt-docs to pick up the additional info from the cataglog.json to display in details sections for models.

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: plugins/snowflake/dbt/include/snowflake/macros/catalog.sql [hunk 1]
```diff
@@ -24,7 +24,12 @@
               'Approximate Size' as "stats:bytes:label",
               bytes as "stats:bytes:value",
               'Approximate size of the table as reported by Snowflake' as "stats:bytes:description",
-              (bytes is not null) as "stats:bytes:include"
+              (bytes is not null) as "stats:bytes:include",
+
+              'Last Modified' as "stats:last_modified:label",
+              to_varchar(last_altered, 'mon dd, yyyy-HH24:MI:SS') as "stats:last_modified:value",
```
Comment (jtcohen6) [inline plugins/snowflake/dbt/include/snowflake/macros/catalog.sql:30]: I think we should pick between displaying either:
* a concise, correct value: `to_varchar(convert_timezone('UTC', last_altered), 'yyyy-mm-dd HH24:MI'||' UTC')`
* a user-friendly value: `to_varchar(last_altered, 'mon dd, hh12:mi am')`

I lean toward the former, what do you think?

File: plugins/snowflake/dbt/include/snowflake/macros/catalog.sql [hunk 2]
```diff
@@ -24,7 +24,12 @@
...
+              'The timestamp for last update/change' as "stats:last_modified:description",
+              (last_altered is not null) as "stats:last_modified:include"
```
Comment (jtcohen6) [inline plugins/snowflake/dbt/include/snowflake/macros/catalog.sql:32]: The last-modified timestamp is a little less intuitive for views vs. tables. Views will include most recent data, whether or not the view definition has been modified recently. So really, this value just indicates the last time the view **definition** _potentially_ changed, but says nothing about the underlying data.

I'd expect folks with database experience to be comfortable with that distinction, but that's not a background we can presume all dbt-docs viewers to have.

Additionally, views haven't historically had other stats, so this value is presented alone without context. (Hence the failing integration tests, where views are not expected to have stats.)

What do you think about adjusting this to
```suggestion
              (last_altered is not null and table_type = "BASE TABLE") as "stats:last_modified:include"
```

**Review by jtcohen6 — COMMENTED** [review]
The pro: just by adding this to the catalog query, it automatically shows up in dbt-docs. Neat!

* I left a few comments from a UX perspective
* I think you'll need to amend the integration tests, since they're not expecting an additional stats field
* I just switched the base branch to v0.18.1. Could you merge changes from that branch and add a changelog entry?

**Comment by jtcohen6** [comment]
I think the integration tests are failing because they're running in the Mr-Nobody99 CircleCI org, rather than the Fishtown-Analytics one. The same thing happened over [here](https://github.com/fishtown-analytics/dbt/pull/2741#issuecomment-691201261); I'm not exactly sure of the cause, but it looks like there's a way to force CircleCI to pick up the Fishtown org + requisite env vars.

File: plugins/snowflake/dbt/include/snowflake/macros/catalog.sql [hunk 3]
```diff
@@ -24,7 +24,12 @@
               'Approximate Size' as "stats:bytes:label",
               bytes as "stats:bytes:value",
               'Approximate size of the table as reported by Snowflake' as "stats:bytes:description",
-              (bytes is not null) as "stats:bytes:include"
+              (bytes is not null) as "stats:bytes:include",
+
+              'Last Modified' as "stats:last_modified:label",
+              to_varchar(convert_timezone('UTC', last_altered), 'yyyy-mm-dd HH24:MI'||'UTC') as "stats:last_modified:value",
+              'The timestamp for last update/change' as "stats:last_modified:description",
+              (last_altered is not null and table_type="BASE_TABLE") as "stats:last_modified:include"
```
Comment (jtcohen6) [inline plugins/snowflake/dbt/include/snowflake/macros/catalog.sql:32]: Double quotes on Snowflake indicates a quoted identifier (i.e. column name), you need single quotes here to indicate a string literal.
```suggestion
	              (last_altered is not null and table_type='BASE TABLE') as "stats:last_modified:include"
```

**Comment by jtcohen6** [comment]
@Mr-Nobody99 I think we caught you amidst the branch switching. Sorry about that. You could try rebasing, but given that your changes were self-contained, I'd recommend resetting your local branch against `dev/0.18.1`, re-adding your changes (snowflake catalog query + changelog), and force-pushing your remote branch.

If that's not something you feel up to, let me know and I can try to help out.

**Comment by Mr-Nobody99** [comment]
@jtcohen6 Thanks for the advice, I hard reset my branch against the upstream dev/0.8.1 and it seems to pass more of the tests now than previously. The value in the test log seems correct however it is still failing, If you have any insight on the cause of the test failure that would be much appreciated.

**Comment by beckjake** [comment]
That snowflake test failure is because the catalog tests haven't been updated to include your changes! In `test/integration/029_docs_generate_tests/test_docs_generate.py`, you'll likely need to update the `_snowflake_stats` method to include a `last_modified` field that matches your changes. For the `value` field you'll probably just want to set `ANY`.

**Comment by Mr-Nobody99** [comment]
@beckjake That was the thing, much appreciated.

**Review by jtcohen6 — APPROVED** [review]
Thanks for the contribution @Mr-Nobody99!

**Comment by Mr-Nobody99** [comment]
@jtcohen6 happy to help 👍

**Pull request merged**